AppKit/CPTrackingArea.j

```diff
@@ -0,0 +1,141 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+@import <Foundation/Foundation.j>
+@import "CPView.j"
+
+@typedef CPTrackingAreaOptions
+CPTrackingMouseEnteredAndExited     = 1 << 1,
+CPTrackingMouseMoved                = 1 << 2,
+CPTrackingCursorUpdate              = 1 << 3,
+CPTrackingActiveWhenFirstResponder  = 1 << 4,
+CPTrackingActiveInKeyWindow         = 1 << 5,
+CPTrackingActiveInActiveApp         = 1 << 6, // both mean the same in cappuccino context
+CPTrackingActiveAlways              = 1 << 6,
+CPTrackingAssumeInside              = 1 << 7,
+CPTrackingInVisibleRect             = 1 << 8,
+CPTrackingEnabledDuringMouseDrag    = 1 << 9;
+
+var CPTrackingAreaRectKey            = @"CPTrackinkAreaRectKey",
+    CPTrackingAreaOptionsKey         = @"CPTrackingAreaOptionsKey",
+    CPTrackingAreaOwnerKey           = @"CPTrackingAreaOwnerKey",
+    CPTrackingAreaUserInfoKey        = @"CPTrackingAreaUserInfoKey",
+    CPTrackingAreaReferencingViewKey = @"CPTrackingAreaReferencingViewKey";
+
+
+/*!
+ @ingroup appkit
+ 
+ A CPTrackingArea defines a region of view that generates mouse-tracking and
+ cursor-update events when the mouse is over that region.
+ */
+@implementation CPTrackingArea : CPObject
+{
+    CGRect                  _rect           @accessors(getter=rect);
+    CPTrackingAreaOptions   _options        @accessors(getter=options);
+    id                      _owner          @accessors(getter=owner);
+    CPDictionary            _userInfo       @accessors(getter=userInfo);
+    
+    CPView                  _referencingView;
+}
+
+
+#pragma mark -
+#pragma mark Initialization
+
+/*! Initializes and returns an object defining a region of a view to receive mouse-tracking events, mouse-moved events, cursor-update events, or possibly all these events.
+ */
+- (CPTrackingArea)initWithRect:(CGRect)aRect options:(CPTrackingAreaOptions)options owner:(id)owner userInfo:(CPDictionary)userInfo
+{
+    if (self = [super init])
+    {
+        if (!(options > 0))
+            [CPException raise:CPInternalInconsistencyException reason:"Invalid CPTrackingArea options"];
+
+        // Check options:
+        // - at least one in CPTrackingMouseEnteredAndExited, CPTrackingMouseMoved, CPTrackingCursorUpdate
+        // - exactly  one in CPTrackingActiveWhenFirstResponder, CPTrackingActiveInKeyWindow, CPTrackingActiveAlways
```

Comment:
Interesting... the old docs say "one of the following", the new docs say "you can configure the scope of activity for mouse tracking". In any case, I agree that it's best to force only one.